src/components/translation-builder/translation-builder.component.tsx

```diff
@@ -0,0 +1,193 @@
+import React, { useState, useEffect, useCallback } from 'react';
+import { useTranslation } from 'react-i18next';
+import {
+  Button,
+  InlineLoading,
+  InlineNotification,
+  IconButton,
+  Tabs,
+  Tab,
+  TabList,
+  Dropdown,
+  ComposedModal,
+} from '@carbon/react';
+import { TrashCan, Add, Download, Edit } from '@carbon/react/icons';
+import { useParams } from 'react-router-dom';
+import styles from './translation-builder.module.scss';
+import { showSnackbar } from '@openmrs/esm-framework';
+import EditTranslationModal from './edit-translation.modal';
+
+interface TranslationBuilderProps {
+  formSchema: any;
+  onUpdateSchema: (updatedSchema: any) => void;
+}
+
+function extractTranslatableStrings(form: any): Record<string, string> {
+  const result: Record<string, string> = {};
+  if (form.pages) {
+    form.pages.forEach((page: any) => {
+      if (page.label) result[page.label] = page.label;
+      if (page.sections) {
+        page.sections.forEach((section: any) => {
+          if (section.label) result[section.label] = section.label;
+          if (section.questions) {
+            section.questions.forEach((question: any) => {
+              if (question.label) result[question.label] = question.label;
+              if (question.questions) {
+                question.questions.forEach((subQuestion: any) => {
+                  if (subQuestion.label) result[subQuestion.label] = subQuestion.label;
+                  subQuestion.questionOptions?.answers?.forEach((answer: any) => {
+                    if (answer.label) result[answer.label] = answer.label;
+                  });
+                });
+              }
+              question.questionOptions?.answers?.forEach((answer: any) => {
+                if (answer.label) result[answer.label] = answer.label;
+              });
+            });
+          }
+        });
+      }
+    });
+  }
+  return result;
+}
+
+const TranslationBuilder: React.FC<TranslationBuilderProps> = ({ formSchema, onUpdateSchema }) => {
+  const { t } = useTranslation();
+  const { formId } = useParams<{ formId: string }>();
+
+  const languageOptions = ['French (fr)', 'Spanish (es)', 'German (de)'];
```

Comment:
we can use a object for this and should be stored somewhere else.

```
export const SUPPORTED_LANGUAGES: Language[] = [
  { label: 'English', code: 'en' },
  { label: 'French', code: 'fr' },
  { label: 'Spanish', code: 'es' },
  { label: 'German', code: 'de' },
];
```

Reply:
Also to get the full name of a language code, please use this - http://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DisplayNames/DisplayNames